packages/wallet/src/unipig-wallet.ts

```diff
@@ -0,0 +1,41 @@
+/* External Imports */
+import { DefaultWallet, DefaultWalletDB, BaseDB, WalletDB } from '@pigi/core'
+
+/* Internal Imports */
+import { Address, Balances, TransactionReceipt, MockRollupClient } from '.'
+
+const KEYSTORE_BUCKET = 0
+const ROLLUP_BUCKET = 1
+
+/*
+ * The UnipigWallet class can be used to interact with the OVM and
+ * all the L2s under it.
+ */
+export class UnipigWallet extends DefaultWallet {
+  private db: BaseDB
+  public rollup: MockRollupClient
+
+  constructor(db: BaseDB) {
+    // Set up the keystore db
+    const keystoreBucket = db.bucket(Buffer.from([KEYSTORE_BUCKET]))
+    const keystoreDB: WalletDB = new DefaultWalletDB(keystoreBucket)
+    super(keystoreDB)
+
+    // Set up the rollup client db
+    const rollupBucket = db.bucket(Buffer.from([ROLLUP_BUCKET]))
+    this.rollup = new MockRollupClient(rollupBucket, this.sign)
+
+    // Save a reference to our db
+    this.db = db
+  }
+
+  public async sign(address: string, message: string): Promise<string> {
+    // Mock the signature for now
+    return address
```

Comment:
Maybe inject a `SignatureProvider` in the constructor, and then this doesn't have to be mocked here? Can always inject a mocked `SignatureProvider` for the same outcome.